### PR TITLE
Get cmd line build working

### PR DESCRIPTION
**Pull request by robward-ms**
### Description

Get cmd line build working:
1.) Switch to ps1 script to execute build, tests, and display results.
2.) Delete unused MSBuild files.
3.) Remove xunit.execution.dotnet from the AspNet UnitTest project since it's desktop, not dotnet.
4.) Repath UT and E2E output to AspNet and AspNetCore subfolders.
5.) Fix issue in security helper where stdErr was not checked.
6.) Modify PublicAPi test to write directly to file instead of capturing StdOut, which caused the
    test to fail occasionally by adding other Console.Write()'s to the .out file.
7.) Fix duplicate test case in ODataRawValueSerializerTests.cs

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+}
+elseif ($args[0] -match 'quick' -or ($args[0] -match '-q')) 
+{
+    $TestType = "Quick"
+}
+elseif ($args[0] -match 'DisableSkipStrongName')
+{
+    $TestType = "DisableSkipStrongName"
+}
+elseif ($args[0] -match 'EnableSkipStrongName')
```
Comment (robward-ms) [inline build.ps1]: There is a downside to this version of 'EnableSkipStrongName' and the VSTS build: running this script with 'EnableSkipStrongName' causes all projects to be rebuilt. ODL gets around this by running 'EnableSkipStrongName' as a machine prep step but WebApi has it as part of the build.

The VSTS build tasks should be changed to stop calling this and instead put it in a machine prep build definition.

File: test/UnitTest/Microsoft.AspNet.OData.Test/Microsoft.AspNet.OData.Test.csproj [hunk 2]
```diff
@@ -10,8 +10,8 @@
     <RootNamespace>Microsoft.AspNet.OData.Test</RootNamespace>
     <AssemblyName>Microsoft.AspNet.OData.Test</AssemblyName>
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <OutputPath>..\..\..\bin\$(Configuration)\UnitTest\</OutputPath>
-    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\..\</SolutionDir>
+    <OutputPath>$(WebStackRootPath)bin\$(Configuration)\UnitTest\AspNet\</OutputPath>
```
Comment (robward-ms) [inline test/UnitTest/Microsoft.AspNet.OData.Test/Microsoft.AspNet.OData.Test.csproj]: This causes the VSTS build to no longer find the test binary. I modified the VSTS task to work for the old and new path by adding \**\.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj [hunk 3]
```diff
@@ -2,66 +2,20 @@
...
-    <FileAlignment>512</FileAlignment>
-    <SccProjectName>SAK</SccProjectName>
-    <SccLocalPath>SAK</SccLocalPath>
-    <SccAuxPath>SAK</SccAuxPath>
-    <SccProvider>SAK</SccProvider>
-    <OutputPath>..\..\..\bin\$(Configuration)\E2ETest\</OutputPath>
-    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\src\</SolutionDir>
-    <RestorePackages>true</RestorePackages>
-    <TargetFrameworkProfile />
+    <OutputPath>$(WebStackRootPath)bin\$(Configuration)\E2ETest\AspNet\</OutputPath>
```
Comment (robward-ms) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj]: This causes the VSTS build to no longer find the test binary. I modified the VSTS task to work for the old and new path by adding \**\.

File: test/UnitTest/Microsoft.AspNet.OData.Test/Microsoft.AspNet.OData.Test.csproj (hunk 2 above)
Comment (robward-ms) [reply test/UnitTest/Microsoft.AspNet.OData.Test/Microsoft.AspNet.OData.Test.csproj]: These tests are running in VSTS.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj (hunk 3 above)
Comment (robward-ms) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj]: These tests are NOT running in VSTS.

Comment (robward-ms) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj]: Still not running. Looks like the larger change of adding WebStack.settings.targets and removing other stuff, including AssemblyInfo.cs, may have affected this:

```
2018-05-04T16:41:15.3531177Z Starting test execution, please wait...
2018-05-04T16:41:16.1968761Z Warning: [xUnit.net 00:00:01.5174973] Skipping: Microsoft.Test.E2E.AspNet.OData (could not find dependent assembly 'Microsoft.Test.E2E.AspNet.OData, Version=7.0.0')
2018-05-04T16:41:16.1968761Z 
2018-05-04T16:41:16.2593743Z Warning: No test is available in C:\BuildAgent\_work\77\s\bin\Release\E2ETest\AspNet\Microsoft.Test.E2E.AspNet.OData.dll. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again.
```

File: build.ps1 (hunk 1 above)
Comment (robward-ms) [reply build.ps1]: Machine prep definition created, test task modified.

**Comment by robward-ms** [comment]
E2E tests for classic still not running in VSTS.
```

2018-05-06T20:38:26.0282911Z Starting test execution, please wait...
2018-05-06T20:38:26.7626752Z Warning: [xUnit.net 00:00:00.5396433] Skipping: Microsoft.Test.E2E.AspNet.OData (could not find dependent assembly 'Microsoft.Test.E2E.AspNet.OData, Version=7.0.0')
2018-05-06T20:38:26.7626752Z 
2018-05-06T20:38:26.7626752Z Warning: No test is available in C:\BuildAgent\_work\77\s\bin\Release\E2ETest\AspNet\Microsoft.Test.E2E.AspNet.OData.dll. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again.

```

**Comment by robward-ms** [comment]
OK, found the problem. XUnit fails to load the delay-signed binary for E2E AspNetClassic because there is no skip verification entry.

This change is adding WebStack.settings.targets to the E2E for AspNetClassic, which means that the binary is now delay signed. As a result, it needs to have a strong name skip verification entry added. But this change also changed the process for strong name skip verification and it's not running in the build properly. I created the machine prep build definition but it's not running as expected due to the way Nuget is used. 

I'll clean this up tonight to get it ready for checkin.

File: build.ps1 [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+    $DOTNETTEST = $DOTNETDIR + "dotnet.exe"
+}
+else
+{
+   Error("The dotnet CLI must be installed to run any .NET Core tests.")
+   exit
+}
+
+# Other variables
+$FXCOP = $FXCOPDIR + "\FxCopCmd.exe"
```
Comment (biaol-odata) [inline build.ps1]: Should FXCOPDIR for VS2017 be Microsoft Visual Studio\2017\Enterprise\Team Tools\Static Analysis Tools\FxCop? thought?

Comment (robward-ms) [reply build.ps1]: That's a good point. I think the right thing to do is align Fxcop locally with Fxcop in the build. Given that the output is ignored in this script as it is, I'm going to remove it to enable this checkin. A future item can be added to resolve the local and CI versions of Fxcop. Sound good?

Comment (biaol-odata) [reply build.ps1]: Sound reasonable to me, since this shouldn't be a blocker.

File: build.ps1 (hunk 1 above)
Comment (robward-ms) [reply build.ps1]: Machine prep definition created, test task modified. Machine prep now setting up skip string name verification for all binaries.

**Comment by xuzhg** [comment]
@robward-ms thanks. I merged it and I closed my PR: https://github.com/OData/WebApi/pull/1423

**Pull request merged**